### PR TITLE
Backport fPowAllowMinDifficultyBlocks rule to DarkGravityWave

### PR DESCRIPTION
**Pull request by UdjinM6**
This should help extending stuck chains on testnet/devnet which in its turn should help nodes to finish sync process. `000000000000000000000000000000000000000000000000003e9ccfe0e03e01` is the work of the "wrong" chain, so this rule activates there immediately and new blocks with high diff from that chain are going to be rejected by updated nodes. Note, that old nodes are going to reject blocks from updated nodes after the "right" chain reaches this amount of work too. This is a temporary  condition which should be removed when we decide to HF testnet again.

### CODE REVIEW

**Comment by gladcow** [comment]
utACK

File: src/pow.cpp [hunk 1]
```diff
@@ -89,6 +89,19 @@ unsigned int static DarkGravityWave(const CBlockIndex* pindexLast, const Consens
         return bnPowLimit.GetCompact();
     }
 
+    if (params.fPowAllowMinDifficultyBlocks && (
+        // testnet ...
+        (params.hashDevnetGenesisBlock.IsNull() && pindexLast->nChainWork >= UintToArith256(uint256S("0x000000000000000000000000000000000000000000000000003e9ccfe0e03e01"))) ||
```
Comment (nmarley) [inline src/pow.cpp]: Why is this needed when `fPowAllowMinDifficultyBlocks` is set per network?

File: src/pow.cpp [hunk 2]
```diff
@@ -89,6 +89,19 @@ unsigned int static DarkGravityWave(const CBlockIndex* pindexLast, const Consens
...
+        // or devnet
+        !params.hashDevnetGenesisBlock.IsNull())) {
+        // Special difficulty rule for testnet/devnet:
+        // If the new block's timestamp is more than 2* 2.5 minutes
+        // then allow mining of a min-difficulty block.
+        // TODO: "testnet OR devenet" part on next testnet hard-fork
```
Comment (nmarley) [inline src/pow.cpp]: typo: "devenet"

File: src/pow.cpp (hunk 1 above)
Comment (UdjinM6) [reply src/pow.cpp]: Because otherwise almost the whole testnet chain is invalid - bits won't match

Comment (nmarley) [reply src/pow.cpp]: Ok, just read the description and understand why this is here. Might be helpful to put in `// TEMP condition + description` comment in the code so the seemingly "random" target value makes sense.

**Comment by UdjinM6** [comment]
@nmarley Thanks! I'll address your comment in future "cleanup PR" if you don't mind.
